File: phase-1/game-of-life/typescript/src/gameOfLife.ts
import { Grid } from './grid';

export class GameOfLife {
    private grid: Grid;

    constructor(rows: number, cols: number) {
        this.grid = new Grid(rows, cols);
    }

    public start(): void {
        this.grid.print();
        // Logic for game of life will go here
    }
}

File: phase-1/game-of-life/elixir/lib/game_of_life.ex
defmodule GameOfLife do
  @moduledoc """
  Module to start and run the Game of Life.
  """

  def start(rows, cols) do
    grid = GameOfLife.Grid.new(rows, cols)
    GameOfLife.Grid.print(grid)
    # Add the logic for the Game of Life rules here
  end
end
